src/data/augmentations.py

```diff
@@ -579,3 +587,55 @@ def __call__(self, x):
 
     x = tf.pad(x, [[pad_up, pad_down], [pad_left, pad_right], [0,0]], self.padding_strategy)
     return x
+
+
+class Pad:
+  def __init__(self, config):
+    """Pads the image with padding values and strategy.
+
+    Args:
+        padding (int, List[int], Tuple[int]): Depending on the Datatype image is padded with the values accordingly
+        - if int = n, image is padded n times symetrically (i.e padding on top = bottom = left = right = n)
+        - if Tuple/List with length = 2. eg: (m, n), top = bottom = m and left = right = n. 
+        - if Tuple/List with length = 4. eg: (m, n, o, p), top = m, bottom = n, left = o , right = p.
+        padding_strategy (str): tf.pad takes one of these ("CONSTANT", "REFLECT", "SYMMETRIC") values to interpolate the missing pixels
+        pad_constant_value (int): used to fill with constant value when padding_strategy is "CONSTANT"
+
+    returns:
+        (tf.Tensor): image
+    """
+    # check for TypeErrors
+    self.padding = config['padding']
+    if not isinstance(self.padding, (tuple, list, int)):
+      raise TypeError(f"Invalid {type(self.padding)} type. Only tuple, list and int supported")
```

Comment:
maybe convert self.padding to a specific type (tuple for instance) to remove conditions in the code bellow

Reply:
This was added for user convenience. 
If user passes only 1 value(e.g. int) we convert to pad equally on all 4 sides. 
For 2 values then we pad first value to top and bottom and second value to left and right side of the image.
For some reason user wants to pad different values. Padding with 4 different pixels on each side could be an option.

Reply:
I agree. My point was that inside this function, we should convert to tuple to simplify the rest of the code. (without touching the user interface)